### ISSUE TITLE
No visual feedback when starting applications

### ISSUE THREAD

**Issue by agaida** [lxqt/lxqt#1554]
##### Expected Behavior

Maybe a good feature, as users request it.

##### Current Behavior

LXQt gives no feedback like KDE

##### Steps to Reproduce (for bugs)
1. just start a bigger application from them menu or quicklaunch 

Hmm, maybe packagers should recommend some kind of preload daemon in the LXQt metapackages and or the session  to speed up starts of the most used applications.

Link: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=907593

**Comment by tsujan** [comment]
Visual feedback aside, I quote from that report:

> LXDE launches and shutsdown apps a lot quicker than LXQT does.

That isn't true. It depends on *many things*, for example, whether the app is written in GTK+ or Qt. You can't expect pcmanfm-qt to start faster in LXDE or, conversely, pcmanfm in LXQt.

I remember that there were daemons that supposedly made apps' start-up faster when I had a weak computer, 13 years ago (one of them was "intelligent" and remembered which apps were used more often but I didn't feel any difference). I don't remember their names but this isn't related to LXQt.

**Comment by agaida** [comment]
@tsujan - in debian the adaptive  preloader package is called `preload` - and i didn't have the chance to see the behaviour without, because this little daemon is standard in siduction - introduced it eight years ago. And yes, it helps.

**Comment by tsujan** [comment]
Oh, yes, it was `preload`. In those times, I didn't know whether "faster" start-ups were because of `preload` or self-suggestion ;) I didn't inspect more because I got a faster computer soon.

**Comment by agaida** [comment]
ok, you know that i own only an old and slow eight core with nearly no RAM (32G) and only old SSDs (not even fast nvram) - so it speed up things notable.

**Comment by tsujan** [comment]
Wait a minute! The latest version of `preload` is 0.6.4, which goes back to 2009! I wouldn't use it even if it worked.

**Comment by agaida** [comment]
it works - its simple and don't change much over time - make a table of the most used processes and preload things into ram before first usage.

RE swap: ok, i have a small 32G swap partition, most of the time it is not used - except i try to compile QtWebEngine - with -j10 it need approx 48-55G of RAM. So these things depend on the use case. Normally all the things that cause heavy I/O i put into tmpfs

**Comment by tsujan** [comment]
Also see https://wiki.archlinux.org/index.php/Preload

With my 8-core + 16-GiB-RAM + root on sshd, even Firefox starts relatively fast the first time -- the other times, it starts almost instantly. I might retry `preload` or try `gopreload` if I feel brave.

**Comment by tsujan** [comment]
As for heavy I/O in tmpfs, I have:
```
tmpfs           7.8G  2.3M  7.8G   1% /dev/shm
tmpfs           7.8G     0  7.8G   0% /sys/fs/cgroup
tmpfs           7.8G   40M  7.8G   1% /tmp
tmpfs           1.6G   16K  1.6G   1% /run/user/1000
```
`/var/log/journal/` isn't on tmpfs and systemd makes heavy writes there by default. To avoid that as far as possible, in `/etc/systemd/journald.conf`, I set:
```
SystemMaxUse=50M
MaxLevelStore=warning
```

**Comment by agaida** [comment]
Introduced these defaults in 2017 i guess in siduction
```
[Journal]
MaxLevelStore=warning
SystemMaxUse=25M
SystemMaxFileSize=2M
```

**Comment by tsujan** [comment]
> Introduced these defaults in 2017 i guess in siduction

That's very reasonable!

Actually, systemd is unreasonable and we should compensate for that :)

**Comment by agaida** [comment]
systemd is not unreasonable - but they trust in distribution maintainers and user to do the right thing :p - and default to MaxLevelStore=debug is just insane - but thats not the fault of systemd, there should be distribution maintainers to define the distribution defaults at their whish. It is possible :sunglasses:

**Comment by tsujan** [comment]
I didn't know that. Yes, `MaxLevelStore=debug` is insane and I thought it was systemd's default, maybe, because systemd has done many silly things in the past and I didn't expect more.

Anyhow, systemd hasn't had any problem for months. We should celebrate ;)

Back to the main topic:

The visual feedback may be possible in different ways. I have no idea how it could be implemented but think that the task-bar is a good place for it.

**Comment by agaida** [comment]
To be honest - in introduced the configuration files after some years running siduction with systemd because i was tired of user claims. top 3:
* "oh, there is a bug" - "No, thats only a information"
* "oh, there is a bug" - "No, thats only a debug info for people know how to read and interpret ..."
* "Why is this in my journal?" - "Nevermind ..."
This changed after some logs reached some size over the years - biggest thing was 500G - and / was full. After that i thought about some changes, max level was one, file size the secone and log rotation the third. And it speed up things a lot. The best in case of log rotation is: if the files are from the same day/hour  the system has a problem that should be solved :D

**Comment by Vladimir-csp** [comment]
IIRC, "startup notification" is a proper mechanism that addresses visual feedback. 
Quick search reveals a tool that implements it among other things: https://github.com/bbidulock/xdg-launch

**Comment by agaida** [comment]
visual feedback is a nice thing - ok, i never *really* missed that much, but i can understand that users want it. So, if possible without to much downsides or performance penalties, we should implement it.

**Comment by tsujan** [comment]
> So, if possible without to much downsides or performance penalties

I don't think it could have real performance penalties. I know nothing about its implementation yet but haven't seen any drawback where it's implemented.

**Comment by agaida** [comment]
this bug/feature request supercedes #703 - it simply was forgotten.

**Comment by filson367** [comment]
Starting larger applications on older systems can take 5 sec or more until window is presented on screen. Same if cloudressources are involved. Inbetween the user does not know "what is going on." It happens I start the application a second time, ending up with two instances.
I would really appreciate some kind of feedback when starting applications.

**Comment by stefonarch** [comment]
I'm looking a little bit into this, coming to the conclusion that it could be implemented only well on x11 as the protocol works only for X :

https://specifications.freedesktop.org/startup-notification/latest/

This gives a list of which installed apps have the key set and its values:

```
#!/bin/bash
for file in /usr/share/applications/*.desktop; do
    grep -q '^StartupNotify=' "$file" || continue

    #name=$(grep -m1 '^Name=' "$file" | cut -d= -f2-)
    exec_cmd=$(grep -m1 '^Exec=' "$file" | cut -d= -f2-)
    sn=$(grep -m1 '^StartupNotify=' "$file" | cut -d= -f2-)

    echo "$sn:$exec_cmd"
done | sort -r | awk -F: '{printf "%-6s %-30s %s\n", $1, $2, $3}' > list.txt


```

On Wayland only the compositor _can_ handle it, which is what Plasma and Gnome do afaik. 

So the only thing which will work also for Wayland will be a notification window for apps with `StartUpNotify=true` shown for x seconds. I patched libqtxdg and it's not very convenient

1. it will show the notification also when the app (browsers for example) is already open and a link is clicked somewhere. So this would need also a check if the app is already running if possible

2. It  shows notifications only when launched by dbus shortkey, quicklaunch and panel menus  but not when launched from desktop or runner, didn't investigate that.

A setting for it should be added, with timeout as we get no feedback from the app for sure.

**Comment by stefonarch** [comment]
Clicked the wrong button. It ~will~ should show only a notification for the apps that set it to "true".

[show_startupnotify.patch](https://github.com/user-attachments/files/24508964/show_startupnotify.patch)

https://github.com/user-attachments/assets/bb8531ab-89a9-407e-812f-1c5e149d0c03

**Comment by stefonarch** [comment]
While running this patch I noticed some inconsistent behavior, in works also from runner (but not gimp...) and else, my knowledge of `libqtxdg`'s structure is very limited.

We cannot show a spinning cursor on Wayland but a tiny splash but this would need layershell-qt for overlay layer.
